src/test/java/hudson/tasks/JavadocArchiverTest.java

```diff
@@ -30,6 +41,33 @@ private void verify(JavadocArchiver before) throws Exception {
         r.assertEqualDataBoundBeans(before, p.getPublishersList().get(JavadocArchiver.class));
     }
 
+    @Test
+    public void renderJavadoc() throws Exception {
+        FreeStyleProject s8 = setupJavadocProject("javadoc-single-8");
+        FreeStyleProject s11 = setupJavadocProject("javadoc-single-11");
+        FreeStyleProject m8 = setupJavadocProject("javadoc-multi-8");
+        FreeStyleProject m11 = setupJavadocProject("javadoc-multi-11");
+        //r.interactiveBreak();
+        validateJavadocReport(s8);
+        validateJavadocReport(s11);
+        validateJavadocReport(m8);
+        validateJavadocReport(m11);
+    }
+
+    private FreeStyleProject setupJavadocProject(String s) throws Exception {
+        FreeStyleProject p = r.createFreeStyleProject(s);
+        p.setScm(new CopyResourceDir(getClass().getResource(s), "javadoc"));
```

Comment:
For stylistic reasons you should use `JavadocArchiverTest.class`. (A subclass in another package would fail here.)

Reply:
Good point